README.md

```diff
@@ -14,24 +14,44 @@ The image is pushed to the `public.ecr.aws/spacelift/runner-ansible` public repo
 
 Altogether we have 3 flavors of the image:
 
-- `public.ecr.aws/spacelift/runner-ansible` - built on top of the [Spacelift Terraform runner image](https://github.com/spacelift-io/runner-terraform), with Ansible installed.
-- `public.ecr.aws/spacelift/runner-ansible-aws` - built on top of `runner-ansible`, with `boto3` installed.
-- `public.ecr.aws/spacelift/runner-ansible-gcp` - built on top of `runner-ansible`, with `google-auth` installed.
+- `runner-ansible:${ANSIBLE_VERSION}` - built on top of `python:3.12-alpine` base image, with `ansible` and `ansible-runner` installed.
+- `runner-ansible:${ANSIBLE_VERSION}-aws` - built on top of `runner-ansible`, with `boto3` installed.
+- `runner-ansible:${ANSIBLE_VERSION}-gcp` - built on top of `runner-ansible`, with `google-auth` installed.
```

Comment:
building on top of a different image is a breaking change (in case someone used tools from the `runner-terraform` image). have you though about this:
- `runner-ansible:${ANSIBLE_VERSION}` - inherits from `runner-terraform`
- `runner-ansible:${ANSIBLE_VERSION}-azure` - inherits from `runner-terraform:azure-latest`
- `runner-ansible:${ANSIBLE_VERSION}-gcp` - inherits from `runner-terraform:gcp-latest`

That's the cleanest imo

Reply:
I explicitly wanted to avoid inheriting from it here. It does not bundle any terraform binary, but tools and other stuff that are 100% useless for an ansible workflow.

It increases the attack surface for nothing. Not to talk about the time taken to pull a 1.5G image during runs.

> it's a nice move to keep our images consistent

I agree and disagree. That does not make sense to me to have a base image that is inherited by all vendors. Every vendor is too specific. I don't think we'll get any advantages of having a base image. 

Some vendors may want to inherit from a different root image than the one defined in a potential base image.

BC breaks are assumed in those new ansible images, they are not meant to be compatible with the old one.
The main motivation here is to start having smaller images that bundle only what is required, with clean version locking.

> adduser --disabled-password --uid=1983 spacelift

Regarding a common permission model, that's a bit orthogonal here, but I think our actual model is far to be ideal. It is very common to have the default user UID/GID to be set to 1000, having it set to 1983 sounds counter-intuitive to me. We are probably no gonna change that behavior ever, just wanted to give my 50c 😛 

That makes me think that this image runs as `root` by default, which is not ideal. I'll fix it to use `1983` 👍🏻 

I'm not sure it's worth the rigidity of having a base image to enforce a UID that will never change.

> in general, a bit less maintenance

Interesting, for me not inhering from a base image is less maintenance 😅 It's one layer less to take care of.